include/plgd/plgd_time.h

```diff
@@ -0,0 +1,137 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Daniel Adam, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"),
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#ifndef PLGD_TIME_H
+#define PLGD_TIME_H
+
+#include "port/oc_clock.h"
+#include "util/oc_features.h"
+
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef OC_HAS_FEATURE_PLGD_TIME
+
+typedef enum {
+  PLGD_TIME_STATUS_IN_SYNC,
+  PLGD_TIME_STATUS_SYNCING,
```

Comment:
we need to know if time is synced from storage.

`PLGD_TIME_STATUS_IN_SYNC_FROM_STORAGE` ?

Reply:
also getter for `last_synced_time` to public API